pages/index.tsx

```diff
@@ -124,7 +122,8 @@ export default function Home({ deviceType, singleItemDetail }: Props) {
   const handleScanButtonClick = useCallback(() => {
     setUrl(areasURL + "?" + urlParams);
     resetThrottling();
-  }, [resetThrottling, urlParams]);
+    mutate();
```

Comment:
@rokasta12 This will make duplicate requests as mentioned whenever the button is clicked. 

<img width="352" alt="image" src="https://user-images.githubusercontent.com/7020369/218251370-8d67b384-2140-4b01-a48f-0faaf76f867c.png">

First one is triggered by the url change on line 123, which makes a request to the existing URL and that's the redundant request. 

Second one is triggered by the mutate which is the expected one. 

We did a lot of investigation around this yesterday but we could not find an easy way to overcome the issue. So I did a bit of refactoring here https://github.com/acikkaynak/deprem-yardim-frontend/pull/755/files that changes the approach of building the urlParams briefly.

Reply:
Okay now i see duplicate request , i thought it was because cache.  In order to not force for making another request i tried to add a hook for checking value but somehow it doesnt work as expected, also detected some other unnecessary re-renders each second.
<img width="654" alt="image" src="https://user-images.githubusercontent.com/44666921/218259581-0b6922a1-b693-41a5-8bdf-ee21d4317ad8.png">